wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitBasicAuthenticationCredentialsProvider.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.git;
+
+import com.google.inject.Singleton;
+import org.eclipse.che.api.git.exception.GitException;
+import org.eclipse.che.api.git.shared.ProviderInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Credentials provider for Git basic authentication
+ *
+ * @author Yossi Balan
+ */
+@Singleton
+public class GitBasicAuthenticationCredentialsProvider implements CredentialsProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GitBasicAuthenticationCredentialsProvider.class);
+    private static ThreadLocal<UserCredential> currRequestCredentials = new ThreadLocal<>();
+    private static final String GIT_PROJECT_ID = "git-project-id";
+
+    @Override
+    public UserCredential getUserCredential() {
+        UserCredential credentials = currRequestCredentials.get();
+        if (credentials == null) {
+            return null;
+        }
+        LOG.info("Successfully fetched credentials for remote repository");
+        return credentials;
+    }
+
+    @Override
+    public String getId() {
+        return GIT_PROJECT_ID;
+    }
+
+    @Override
+    public boolean canProvideCredentials(String url) {
+        return getUserCredential() != null;
+    }
+
+    @Override
+    public ProviderInfo getProviderInfo() {
+        return null;
```

Comment:
:) Seems that we are not able to return ProviderInfo here. So maybe we can rework ProviderInfo and make authenticateUrl optional. Then we can add constructor only with provider name and `return new ProviderInfo(PROVIDER_ID);` 
Or maybe add description to CredentialsProvider that it can return nullable info. I prefer first way
WDYT?
WDYT?